### ISSUE TITLE
Issues Installing

### ISSUE THREAD

**Issue by ceemoneyx** [NixOS/nix#7343]
Hello, I'm having an issue installing Nix on Linux. Running on a Raspberry Pi.

When I run `curl -L https://nixos.org/nix/install | sh` (with and without sudo)  
it just returns: `sh: sorry, there is no binary distribution of Nix for your platform`

Sorry for the noob question but I can't figure out what to do here haha.

Thanks in advance for any replies

**Comment by bjornfor** [comment]
You can probably find that string in the code tree and extend the error message to include the detected platform, and possibly the list of supported platforms. I think that would be helpful.

**Comment by ceemoneyx** [comment]
Could I trouble you to elaborate a little bit? My knowledge with Linux/Programming isn't very extensive

**Comment by abathur** [comment]
You can run the same uname commands the installer uses and see what they report. I think maybe some but not all pi versions work?

https://github.com/NixOS/nix/blob/master/scripts/install.in#L27-L64

**Comment by ceemoneyx** [comment]
Thanks abathur,

Running `echo "$(uname -s).$(uname -m)"` returns `Linux.armv7l`
So it should work? as I see that included in lines 27-64 of that install code. (or are those items the unsupported versions?)

**Comment by ceemoneyx** [comment]
Looks like I may have to try using a different Linux distribution

**Comment by abathur** [comment]
I've asked on Matrix to see if anyone who has tried nix on a pi has insight

**Comment by zhaofengli** [comment]
You are using a 32-bit OS on your Pi and there are no prebuilt Nix binaries on 32-bit ARM. If you are using a Pi 3 or later, there are two ways to go, depending on your familiarity of Nix:

- [Install the 64-bit version of Raspberry Pi OS](https://www.raspberrypi.com/software/operating-systems/#raspberry-pi-os-64-bit), then install Nix on it. This allows you to use Nix itself while keeping the base system "vanilla." Tutorials targeting Raspberry Pis found online will still work.
- [Install NixOS](https://nixos.wiki/wiki/NixOS_on_ARM). The whole operating system will be configured with Nix expressions.

**Comment by ceemoneyx** [comment]
@zhaofengli I appreciate your response and sparing me the pain haha. I'm running this on a Pi 2B, I'm guessing the 64-bit version won't work?

**Comment by SuperSandro2000** [comment]
Nope, that has armv7 only. You can still try to build an SD card image but that will take a while especially if you don't have an arm64 builder.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
